### PR TITLE
Update tests to match changed op order

### PR DESCRIPTION
**Pull request by jorickert**
Needed for: https://github.com/Xilinx/llvm-project/pull/424

### CODE REVIEW

File: test/Dialect/TMTensor/bufferize.mlir [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
+// CHECK-DAG:       %[[ACC_MEMREF_NEW:.*]] = memref.alloc() : memref<i32>
+// CHECK-DAG:       %[[OUT_TENSOR_NEW:.*]] = bufferization.to_tensor %[[OUT_MEMREF_NEW]] : memref<128xi32>
+// CHECK-DAG:       %[[ACC_TENSOR_NEW:.*]] = bufferization.to_tensor %[[ACC_MEMREF_NEW]] : memref<i32>
 // CHECK:           tm_tensor.scan dimension(0) inclusive(true) ins(%[[IN_MEMREF]] : memref<128xi32>)
 // CHECK-SAME:            outs(%[[OUT_MEMREF_NEW]], %[[ACC_MEMREF_NEW]] : memref<128xi32>, memref<i32>) {
 // CHECK:           ^bb0(%[[OUT_PREV_ELEMENT:.*]]: i32, %[[IN_ELEMENT:.*]]: i32):
 // CHECK:             %[[OUT_CURRENT_ELEMENT:.*]] = arith.addi %[[OUT_PREV_ELEMENT]], %[[IN_ELEMENT]] : i32
 // CHECK:             tm_tensor.yield %[[OUT_CURRENT_ELEMENT]] : i32
 // CHECK:           }
-// CHECK-DAG:           %[[OUT_TENSOR_NEW:.*]] = bufferization.to_tensor %[[OUT_MEMREF_NEW]] : memref<128xi32>
```
Comment (jorickert) [inline test/Dialect/TMTensor/bufferize.mlir:16]: Can someone confirm that moving this above the last use of the memref is valid?

Comment (mgehre-amd) [reply test/Dialect/TMTensor/bufferize.mlir:16]: Looks odd. Is this how the reference look upstream? They have a bump to that LLVM version, right?
Maybe we want to wait until I catched up with torch-mlir bumps?

Comment (jorickert) [reply test/Dialect/TMTensor/bufferize.mlir:16]: Yes, upstream they have the same reference for some of the tests: https://github.com/llvm/torch-mlir/blob/8e0eafd022cd7555c8b58927d3238a7a89e9dbd4/test/Dialect/TMTensor/bufferize.mlir#L10 . I agree that we should try to get to https://github.com/llvm/torch-mlir/pull/3765 in torch-mlir .

**Comment by mgehre-amd** [comment]
Superseeded by https://github.com/Xilinx/torch-mlir/pull/435

**Pull request closed without merging**